### ISSUE TITLE
plotSection(gB_Wad_Wad.misorientation,'axisAngle',[3 5 7 10]*degree,...

### ISSUE THREAD

**Issue by DavidMainprice** [mtex-toolbox/mtex#300]
Using MTEX 5.0.beta.2 & R2016b Mac OS
```matlab
plotSection(gB_Wad_Wad.misorientation,'axisAngle',[3 5 7 10]*degree,...
```
gives this error message
```
Error using nfsftmex
nfsft: Input argument M must be positive.
Error in S2FunHarmonic.quadrature (line 36)
plan = nfsftmex('init_advanced', bw, length(nodes), 1);
Error in vector3d/calcDensity (line 59)
  sF = 4*pi * S2FunHarmonic.quadrature(v,ones(size(v)),varargin{:});

Error in vector3d/smooth (line 36)
    cdata = calcDensity(v(:),S2G,'halfwidth',5*degree,varargin{:});

Error in vector3d/contourf (line 31)
  h = v.smooth(data,'contours',10,'LineStyle','-','LineColor','k',varargin{:});

Error in vector3d/plot (line 64)
    [varargout{1:nargout}] = v.contourf(varargin{:});
Error in axisAngleSections/plotSection (line 104)
      h = plot(v,data{:},'parent',ax,varargin{:},'doNotDraw');
Error in ODFSections/plot (line 73)
    plotSection(oS,mtexFig.gca,s,iv,secData,varargin{:});
Error in orientation/plotSection (line 24)
oS.plot(ori,varargin{:}); 
```
The reason is some of these misorientation angles, like 4 and 7 have no data  and hence it generates the error message 

all the best David

**Comment by ralfHielscher** [comment]
Hi David,

while this is certainly a bug which I should resolve, I think it is not really a good idea to do 
```matlab
plotSection(gB_Wad_Wad.misorientation,'axisAngle',[3 5 7 10]*degree,'contour')
```
since this compute the density for each section seperately. It is much better and much more correct to do
```matlab
mdf = calcODF(gB_Wad_Wad.misorientation)
plotSection(mdf,'axisAngle',[3 5 7 10]*degree,'contour')
```
Ralf.

**Comment by ralfHielscher** [comment]
The point is: what is the density function if we have no sample? Theoretically it should be constant to 1. But this is far away from what you expect.

Ralf.

**Comment by ralfHielscher** [comment]
Hi David,

I think I understood the issue and I think I fixed it in the MTEX 5.0.beta2 I uploaded yesterday. However, the point is that in 
```
plotSection(ori,'contourf')
```
MTEX computes a density function for each section separately. As a consequence, points between the sections are ignored and sections can not be compared which each other.

What is wrong with computing and mdf first? I also could make this implicit. I.e. calling plotSection with option 'contourf' first computes and odf and then sections of it :)

Ralf.

**Comment by DavidMainprice** [comment]
Dear Ralf,

Thanks for tip about gB_Wad_Wad.misorientation which compute the density for each section seperately versus mdf = calcODF(gB_Wad_Wad.misorientation) with the spherical harmonic method using the complete angle range

I will make a demo so users can seen the relative merits.

all the best David
